mods/parsoid.js

```diff
@@ -617,15 +621,54 @@ PSP.callParsoidTransform = function callParsoidTransform(restbase, req, from, to
     var parsoidExtraPath = parsoidExtras.map(encodeURIComponent).join('/');
     if (parsoidExtraPath) { parsoidExtraPath = '/' + parsoidExtraPath; }
 
-    var domain = rp.domain;
-    // Re-map test domain
     var parsoidReq = {
-        uri: this.parsoidHost + '/v2/' + domain + '/'
+        uri: this.parsoidHost + '/v2/' + rp.domain + '/'
             + parsoidTo + parsoidExtraPath,
         headers: { 'content-type': 'application/json' },
         body: req.body
     };
-    return restbase.post(parsoidReq);
+
+    var transformPromise = restbase.post(parsoidReq);
+    if (from === 'wikitext' && to === 'html' && req.body.stash) {
```

Comment:
It might be worth moving this to a separate function, to avoid the stash handling obscuring the common flow too much.